get-release-id.sh

```diff
@@ -7,4 +7,4 @@
 COMMIT_HASH=$1
 source ./resin.env
 
-curl "https://api.$BASE_URL/v2/build?\$select=id,commit_hash&\$filter=application%20eq%20$APP_ID%20and%20commit_hash%20eq%20'$COMMIT_HASH'" -H "Authorization: Bearer $authToken" | jq '.d[0].id' 
+curl "https://api.$BASE_URL/v4/release?\$select=id,commit&\$filter=belongs_to__application%20eq%20$APP_ID%20and%20commit%20eq%20'$COMMIT_HASH'" -H "Authorization: Bearer $authToken" | jq '.d[0].id'
```

Comment:
It might be worth adding a `status: 'success'` qualifier, because in that way we can make sure a user doesn't pin to a failed release - which has happened. It's got protections in the API, but having some client side information might be nice. It's outside of the scope of this PR though so I'm not going to block on it.

Reply:
It seems that the API actually does that check anyway.
On the other hand adding the `status: 'success'` filter sounds like a sensible check and can't think of a reason to not add it.

Reply:
Good idea, done